### PR TITLE
0xMakka / #1183 language switcher

### LINKED ISSUE #1183: [header] Add language switcher functionality (staging only)
The design calls for a language switcher icon in-app.

Screenshot 2023-06-06 at 9 56 26 AM

Need updated designs for the UI when clicking the language switcher button.

Also need design for the homepage.

### PR DESCRIPTION
**Pull request by 0xMakka**
## Description

Adds a language switcher to the homepage header and the header in the carbonmark app.

## Related Ticket

Resolves #1183 

## Checklist

- [X] I have run `npm run build-all` without errors
- [X] I have formatted JS and TS files with `npm run format-all`

### CODE REVIEW

**Review by jabby09 — APPROVED** [review]

**Comment by jabby09** [comment]
@0xMakka @Atmosfearful do we agree this is complete enough that the translators can work off the PR link to finish the translation work?

File: carbonmark/components/ChangeLanguageButton/index.tsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    setLocale(locale);
+    selectLocale(locale);
+  }, []);
+
+  const selectLocale = async (locale: string) => {
+    const messages = await loadTranslation(locale);
+    await activateLocale(locale, messages);
+    window?.localStorage?.setItem("locale", locale);
+    setLocale(locale);
+  };
```
Comment (Atmosfearful) [inline carbonmark/components/ChangeLanguageButton/index.tsx:23]: This is the approach one would take for a single page app where locales are managed client-side (like our `app` package), but Carbonmark can use nextjs builtins. See `site` for an example:

https://github.com/KlimaDAO/klimadao/blob/9d32d9678e53dea2566fd45ca3b61b4283fc6803/site/components/ChangeLanguageButton/index.tsx#L13

**Review by Atmosfearful — CHANGES_REQUESTED** [review]
@jabby09 Yes Pseudo is working so we can use this carbonmark preview link to start identifying missing translations. However I'm realizing that it's a bit cumbersome to have a non-dev identify the strings, as theyd have to screenshot them and have a dev go in to tag the manyways.

So it probably would go faster if we just have @0xMakka dig through all the views and slap `<Trans>` or `t` tags on everything he can find. Let's just do that in this PR.

Please note that we can't translate CMS content, so project names and descriptions will be untranslated.

Also for you to note @0xMakka we are not using `ids` anymore except for very long strings. This reduces a ton of boilerplate: so `<Trans>My String</Trans>` and 
```
t`My Message`
```
is all you need to do :)

I'm gonna add a section in our github wiki on translation best practices and update this comment with a link to it...

**Comment by jabby09** [comment]
> It's hard to see but the bg color doesnt match the button links

nice catch, I did not see that!

**Comment by 0xMakka** [comment]
> It's hard to see but the bg color doesnt match the button links 
> image
> 

Thanks for the feedback @Atmosfearful - eagle eye 😎, great catch I didn't notice this either.

And good to know with using the nextjs built ins, will make the changes to this 👍.

**Review by Atmosfearful — APPROVED** [review]

**Pull request merged**